packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/types.ts

```diff
@@ -0,0 +1,62 @@
+export interface ProviderMeta {
+  ProviderName: string;
+  authorize_scopes: string;
+  AttributeMapping: { [key: string]: string | undefined };
+}
+
+interface ProviderCredsBase {
+  ProviderName: string;
+  client_id: string;
+}
+
+export interface DefaultProviderCreds extends ProviderCredsBase {
+  client_secret: string;
+}
+
+export interface AppleProviderCreds extends ProviderCredsBase {
+  key_id: string;
+  private_key: string;
+  team_id: string;
+}
+
+export type ProviderCreds = DefaultProviderCreds | AppleProviderCreds;
+
+export interface FacebookProviderParameters {
+  facebookAuthorizeScopes?: string;
+  facebookAppIdUserPool?: string;
+  facebookAppSecretUserPool?: string;
+}
+
+export interface GoogleProviderParameters {
+  googleAuthorizeScopes?: string;
+  googleAppIdUserPool?: string;
+  googleAppSecretUserPool?: string;
+}
+
+export interface LoginWitAmazonProviderParameters {
+  loginwithamazonAuthorizeScopes?: string;
+  loginwithamazonAppIdUserPool?: string;
+  loginwithamazonAppSecretUserPool?: string;
+}
+
+export interface SignInWithAppleProviderParameters {
+  signinwithappleAuthorizeScopes?: string;
+  signinwithappleClientIdUserPool?: string;
```

Comment:
casing of property names in this file is all over the place. why is that?

Reply:
I'm following what already exists: https://github.com/aws-amplify/amplify-cli/blob/dev/packages/amplify-category-auth/src/provider-utils/awscloudformation/service-walkthrough-types/cognito-user-input-types.ts#L79-L91